### ISSUE TITLE
logging setup fails when passing a relative path to path.settings

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#6369]
from https://discuss.elastic.co/t/statuslogger-error-while-starting-logstash-5-but-indexting-working/64686

-----

When I start logstash using the command;

`./logstash --path.settings=../config/ -f ../logstash.conf`
I am getting the error;

```
ERROR StatusLogger Unable to access file://../config/log4j2.properties
```

But after sometime, the fetching of data from jdbc (that I have defined in conf file) completes and it indexes values to easticsearch. Why is this happening, and how can I fix this? I have set 0770 for all files in config folder.

----

It seems path.config can't be a relative path, since the same string is passed to the code that loads the log4.properties file:

```
/tmp/logstash-5.0.0/bin % ./logstash --path.settings=../config/ -f ../cfg
ERROR StatusLogger Unable to access file://../config/log4j2.properties
 java.net.UnknownHostException: ..
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
	at java.net.Socket.connect(Socket.java:589)
	at java.net.Socket.connect(Socket.java:538)
	at sun.net.ftp.impl.FtpClient.doConnect(FtpClient.java:957)
	at sun.net.ftp.impl.FtpClient.tryConnect(FtpClient.java:917)
	at sun.net.ftp.impl.FtpClient.connect(FtpClient.java:1012)
	at sun.net.ftp.impl.FtpClient.connect(FtpClient.java:998)
	at sun.net.www.protocol.ftp.FtpURLConnection.connect(FtpURLConnection.java:294)
	at sun.net.www.protocol.ftp.FtpURLConnection.getInputStream(FtpURLConnection.java:393)
	at java.net.URL.openStream(URL.java:1038)
         ...
ERROR StatusLogger Unable to access file://../config/log4j2.properties
 java.net.UnknownHostException: ..
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
	at java.net.Socket.connect(Socket.java:589)
	at java.net.Socket.connect(Socket.java:538)
	at sun.net.ftp.impl.FtpClient.doConnect(FtpClient.java:957)
	at sun.net.ftp.impl.FtpClient.tryConnect(FtpClient.java:917)
	at sun.net.ftp.impl.FtpClient.connect(FtpClient.java:1012)
	at sun.net.ftp.impl.FtpClient.connect(FtpClient.java:998)
	at sun.net.www.protocol.ftp.FtpURLConnection.connect(FtpURLConnection.java:294)
	at sun.net.www.protocol.ftp.FtpURLConnection.getInputStream(FtpURLConnection.java:393)
	at java.net.URL.openStream(URL.java:1038)
         ...
	at org.jruby.Ruby.runScript(Ruby.java:857)
	at org.jruby.Ruby.runScript(Ruby.java:850)
	at org.jruby.Ruby.runNormally(Ruby.java:729)
	at org.jruby.Ruby.runFromMain(Ruby.java:578)
	at org.jruby.Main.doRunFromMain(Main.java:393)
	at org.jruby.Main.internalRun(Main.java:288)
	at org.jruby.Main.run(Main.java:217)
	at org.jruby.Main.main(Main.java:197)
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console.
Sending Logstash logs to /tmp/logstash-5.0.0/logs which is now configured via log4j2.properties.
11:12:31.239 [[main]-pipeline-manager] INFO  logstash.pipeline - Starting pipeline {"id"=>"main", "pipeline.workers"=>4, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>500}
11:12:31.255 [[main]-pipeline-manager] INFO  logstash.pipeline - Pipeline main started
11:12:31.326 [Api Webserver] INFO  logstash.agent - Successfully started Logstash API endpoint {:port=>9600}
{
          "epoc" => 1478085151,
      "sequence" => 0,
    "@timestamp" => 2016-11-02T11:12:31.487Z,
      "@version" => "1",
          "host" => "Joaos-MBP-5.lan",
       "message" => "Hello world!"
}
11:12:34.279 [LogStash::Runner] WARN  logstash.agent - stopping pipeline {:id=>"main"}
/tmp/logstash-5.0.0/bin %
```

The two following log messages are contradictory so this is a bug:

```
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console.
Sending Logstash logs to /tmp/logstash-5.0.0/logs which is now configured via log4j2.properties.
```

**Comment by jsvd** [comment]
NOTE: reports also appeared that on windows the same error happens with full paths on windows

```
E:\Software\logstash-5.0.0\bin\logstash.bat -f E:\Software\logstash-5.0.0\logstash.conf -l E:\Logs\Logstash\logstash.log
ERROR StatusLogger Unable to access file://E:/Software/logstash-5.0.0/config/log4j2.properties
java.net.UnknownHostException: E
    at java.net.AbstractPlainSocketImpl.connect(Unknown Source)
    at java.net.PlainSocketImpl.connect(Unknown Source)
    at java.net.Socket.connect(Unknown Source)
    at java.net.Socket.connect(Unknown Source)
...

**Comment by MerlinFeng** [comment]
Dose this question has been solved? I have a trouble with it!

**Comment by jayaprakashdeav** [comment]
I am facing the problem  below Please help me on this.

PS C:\> cd "C:\Program Files\logstash"

PS C:\Program Files\logstash> bin\logstash.bat -f logstash.conf
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console.
An unexpected error occurred! :error => bad URI(is not URI?): file:///C:/Program Files/logstash/config/log4j2.properties, :backtrace => ["C:/Program Files/logstash/vendor/jruby/lib
/ruby/1.9/uri/common.rb:176:in `split'", "C:/Program Files/logstash/vendor/jruby/lib/ruby/1.9/uri/common.rb:210:in `parse'", "C:/Program Files/logstash/vendor/jruby/lib/ruby/1.9/ur
i/common.rb:747:in `parse'", "C:/Program Files/logstash/vendor/jruby/lib/ruby/1.9/uri/common.rb:994:in `URI'", "C:/Program Files/logstash/logstash-core/lib/logstash/logging/logger.
rb:77:in `initialize'", "org/jruby/ext/thread/Mutex.java:149:in `synchronize'", "C:/Program Files/logstash/logstash-core/lib/logstash/logging/logger.rb:75:in `initialize'", "C:/Pro
gram Files/logstash/logstash-core/lib/logstash/runner.rb:229:in `execute'", "C:/Program Files/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.5/lib/clamp/command.rb:67:in `run'", "
C:/Program Files/logstash/logstash-core/lib/logstash/runner.rb:209:in `run'", "C:/Program Files/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.5/lib/clamp/command.rb:132:in `run'"
, "C:\\Program Files\\logstash\\lib\\bootstrap\\environment.rb:71:in `(root)'"]

**Comment by jitendrasoni** [comment]
I am facing the same problem, I have created and tried with some filter.config but it's was always showing the same exception.

ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console.

Does this means logstash not working on any window machines

**Comment by jakelandis** [comment]
I just this against 5.6.3 on Windows 10, and relative paths work ! .... unless you have a space in your relative path, then it does not :(

The work around is to use a path that does not include spaces. 

Also, the `ERROR StatusLogger No log4j2 configuration file found.` is un-related, a benign error, and no longer appears in newer versions.

**Comment by jakelandis** [comment]
Closing this issue as fixed as of 5.6.3 (tested, but maybe fixed earlier ?) and the space issue is duplicate of https://github.com/elastic/logstash/issues/6426.
